Remove atlas install script.

galv opened the pull request:
@danpovey In my typical style, I didn't do exactly what you suggested. Instead of committing cblas.h and clapack.h, as found in the atlas 3.10 distribution, I changed the tools setup to check whether libatlas-dev is installed on debian distros, and changed configure to link to the packaged /usr/include/atlas-base/ with those .h files. It's not backwards compatible if a user doesn't already have libatlas-dev. If this is absolutely not okay with you, I can redo this to do what you suggested with committing the two files instead. If you're okay with this, I'd like to test a clean install on centos first before committing.

Commit message is below:

===========

It's broken right now anyway and confuses users. We expect users to
install ATLAS via their distribution's package manager or to know what
they're doing if they're not going to do that.

Also, fix tools/extras/check_dependencies.sh to check for libatlas-dev
in debian-based distributions. This contains the cblas.h and clapack.h
header files, so we nol onger need the tools/ATLAS/include workaround.

This change is not backwards-compatible. When users update their code to
this commit, the include path for atlas changes to the include path
provided by libatlas-dev, instead of Kaldi's tools/ATLAS/include (which
is removed). If a user doesn't have libatlas-dev on debian, their build
will break.

danpovey commented:
Sorry, I'm not OK with this approach.  I can understand why you have a
strong preference to using the system-provided headers on Debian, as it's
cleaner, but I'm looking at this from the point of view from the user
experience.  Your approach will break a build that's working.
The fact that it's "nicer" to use the headers from the system is of no
concern to the user.
Even aside from the fact that it breaks back compatibility on Debian, your
approach has danger because changes to the build process are hard to debug
(since we don't have access to all the systems people use).  So I prefer to
be very conservative in changing anything that touches the build process.


On Wed, Oct 25, 2017 at 3:02 AM, Daniel Galvez <notifications@github.com>
wrote:

> @danpovey <https://github.com/danpovey> In my typical style, I didn't do
> exactly what you suggested. Instead of committing cblas.h and clapack.h, as
> found in the atlas 3.10 distribution, I changed the tools setup to check
> whether libatlas-dev is installed on debian distros, and changed configure
> to link to the packaged /usr/include/atlas-base/ with those .h files. It's
> not backwards compatible if a user doesn't already have libatlas-dev. If
> this is absolutely not okay with you, I can redo this to do what you
> suggested with committing the two files instead. If you're okay with this,
> I'd like to test a clean install on centos first before committing.
>
> Commit message is below:
>
> ===========
>
> It's broken right now anyway and confuses users. We expect users to
> install ATLAS via their distribution's package manager or to know what
> they're doing if they're not going to do that.
>
> Also, fix tools/extras/check_dependencies.sh to check for libatlas-dev
> in debian-based distributions. This contains the cblas.h and clapack.h
> header files, so we nol onger need the tools/ATLAS/include workaround.
>
> This change is not backwards-compatible. When users update their code to
> this commit, the include path for atlas changes to the include path
> provided by libatlas-dev, instead of Kaldi's tools/ATLAS/include (which
> is removed). If a user doesn't have libatlas-dev on debian, their build
> will break.
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/kaldi-asr/kaldi/pull/1972
> Commit Summary
>
>    - Remove atlas install script.
>
> File Changes
>
>    - *M* src/configure
>    <https://github.com/kaldi-asr/kaldi/pull/1972/files#diff-0> (41)
>    - *M* tools/Makefile
>    <https://github.com/kaldi-asr/kaldi/pull/1972/files#diff-1> (17)
>    - *M* tools/extras/check_dependencies.sh
>    <https://github.com/kaldi-asr/kaldi/pull/1972/files#diff-2> (5)
>    - *D* tools/extras/install_atlas.sh
>    <https://github.com/kaldi-asr/kaldi/pull/1972/files#diff-3> (54)
>    - *D* tools/install_atlas.sh
>    <https://github.com/kaldi-asr/kaldi/pull/1972/files#diff-4> (1)
>
> Patch Links:
>
>    - https://github.com/kaldi-asr/kaldi/pull/1972.patch
>    - https://github.com/kaldi-asr/kaldi/pull/1972.diff
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/1972>, or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADJVu0HcP80bll5EDix0Bw554S0N-OWRks5svtz7gaJpZM4QFhjO>
> .
>

galv commented:
I will defer to your opinion and redo this to commit the cblas.h and
clapack.h headers, with minimal changes to the build-related files.

I'll try to get it done by tonight.

On Wed, Oct 25, 2017 at 10:38 AM, Daniel Povey <notifications@github.com>
wrote:

> Sorry, I'm not OK with this approach. I can understand why you have a
> strong preference to using the system-provided headers on Debian, as it's
> cleaner, but I'm looking at this from the point of view from the user
> experience. Your approach will break a build that's working.
> The fact that it's "nicer" to use the headers from the system is of no
> concern to the user.
> Even aside from the fact that it breaks back compatibility on Debian, your
> approach has danger because changes to the build process are hard to debug
> (since we don't have access to all the systems people use). So I prefer to
> be very conservative in changing anything that touches the build process.
>
>
> On Wed, Oct 25, 2017 at 3:02 AM, Daniel Galvez <notifications@github.com>
> wrote:
>
> > @danpovey <https://github.com/danpovey> In my typical style, I didn't do
> > exactly what you suggested. Instead of committing cblas.h and clapack.h,
> as
> > found in the atlas 3.10 distribution, I changed the tools setup to check
> > whether libatlas-dev is installed on debian distros, and changed
> configure
> > to link to the packaged /usr/include/atlas-base/ with those .h files.
> It's
> > not backwards compatible if a user doesn't already have libatlas-dev. If
> > this is absolutely not okay with you, I can redo this to do what you
> > suggested with committing the two files instead. If you're okay with
> this,
> > I'd like to test a clean install on centos first before committing.
> >
> > Commit message is below:
> >
> > ===========
> >
> > It's broken right now anyway and confuses users. We expect users to
> > install ATLAS via their distribution's package manager or to know what
> > they're doing if they're not going to do that.
> >
> > Also, fix tools/extras/check_dependencies.sh to check for libatlas-dev
> > in debian-based distributions. This contains the cblas.h and clapack.h
> > header files, so we nol onger need the tools/ATLAS/include workaround.
> >
> > This change is not backwards-compatible. When users update their code to
> > this commit, the include path for atlas changes to the include path
> > provided by libatlas-dev, instead of Kaldi's tools/ATLAS/include (which
> > is removed). If a user doesn't have libatlas-dev on debian, their build
> > will break.
> > ------------------------------
> > You can view, comment on, or merge this pull request online at:
> >
> > https://github.com/kaldi-asr/kaldi/pull/1972
> > Commit Summary
> >
> > - Remove atlas install script.
> >
> > File Changes
> >
> > - *M* src/configure
> > <https://github.com/kaldi-asr/kaldi/pull/1972/files#diff-0> (41)
> > - *M* tools/Makefile
> > <https://github.com/kaldi-asr/kaldi/pull/1972/files#diff-1> (17)
> > - *M* tools/extras/check_dependencies.sh
> > <https://github.com/kaldi-asr/kaldi/pull/1972/files#diff-2> (5)
> > - *D* tools/extras/install_atlas.sh
> > <https://github.com/kaldi-asr/kaldi/pull/1972/files#diff-3> (54)
> > - *D* tools/install_atlas.sh
> > <https://github.com/kaldi-asr/kaldi/pull/1972/files#diff-4> (1)
> >
> > Patch Links:
> >
> > - https://github.com/kaldi-asr/kaldi/pull/1972.patch
> > - https://github.com/kaldi-asr/kaldi/pull/1972.diff
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/kaldi-asr/kaldi/pull/1972>, or mute the thread
> > <https://github.com/notifications/unsubscribe-auth/
> ADJVu0HcP80bll5EDix0Bw554S0N-OWRks5svtz7gaJpZM4QFhjO>
> > .
> >
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/1972#issuecomment-339410451>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AEi_ULUFbQ-VfDWEtelovF-SHMzMDL9Mks5sv3IKgaJpZM4QFhjO>
> .
>



-- 
Daniel Galvez

danpovey commented:
thanks!

On Wed, Oct 25, 2017 at 2:22 PM, Daniel Galvez <notifications@github.com>
wrote:

> I will defer to your opinion and redo this to commit the cblas.h and
> clapack.h headers, with minimal changes to the build-related files.
>
> I'll try to get it done by tonight.
>
> On Wed, Oct 25, 2017 at 10:38 AM, Daniel Povey <notifications@github.com>
> wrote:
>
> > Sorry, I'm not OK with this approach. I can understand why you have a
> > strong preference to using the system-provided headers on Debian, as it's
> > cleaner, but I'm looking at this from the point of view from the user
> > experience. Your approach will break a build that's working.
> > The fact that it's "nicer" to use the headers from the system is of no
> > concern to the user.
> > Even aside from the fact that it breaks back compatibility on Debian,
> your
> > approach has danger because changes to the build process are hard to
> debug
> > (since we don't have access to all the systems people use). So I prefer
> to
> > be very conservative in changing anything that touches the build process.
> >
> >
> > On Wed, Oct 25, 2017 at 3:02 AM, Daniel Galvez <notifications@github.com
> >
> > wrote:
> >
> > > @danpovey <https://github.com/danpovey> In my typical style, I didn't
> do
>
> > > exactly what you suggested. Instead of committing cblas.h and
> clapack.h,
> > as
> > > found in the atlas 3.10 distribution, I changed the tools setup to
> check
> > > whether libatlas-dev is installed on debian distros, and changed
> > configure
> > > to link to the packaged /usr/include/atlas-base/ with those .h files.
> > It's
> > > not backwards compatible if a user doesn't already have libatlas-dev.
> If
> > > this is absolutely not okay with you, I can redo this to do what you
> > > suggested with committing the two files instead. If you're okay with
> > this,
> > > I'd like to test a clean install on centos first before committing.
> > >
> > > Commit message is below:
> > >
> > > ===========
> > >
> > > It's broken right now anyway and confuses users. We expect users to
> > > install ATLAS via their distribution's package manager or to know what
> > > they're doing if they're not going to do that.
> > >
> > > Also, fix tools/extras/check_dependencies.sh to check for libatlas-dev
> > > in debian-based distributions. This contains the cblas.h and clapack.h
> > > header files, so we nol onger need the tools/ATLAS/include workaround.
> > >
> > > This change is not backwards-compatible. When users update their code
> to
> > > this commit, the include path for atlas changes to the include path
> > > provided by libatlas-dev, instead of Kaldi's tools/ATLAS/include (which
> > > is removed). If a user doesn't have libatlas-dev on debian, their build
> > > will break.
> > > ------------------------------
> > > You can view, comment on, or merge this pull request online at:
> > >
> > > https://github.com/kaldi-asr/kaldi/pull/1972
> > > Commit Summary
> > >
> > > - Remove atlas install script.
> > >
> > > File Changes
> > >
> > > - *M* src/configure
> > > <https://github.com/kaldi-asr/kaldi/pull/1972/files#diff-0> (41)
> > > - *M* tools/Makefile
> > > <https://github.com/kaldi-asr/kaldi/pull/1972/files#diff-1> (17)
> > > - *M* tools/extras/check_dependencies.sh
> > > <https://github.com/kaldi-asr/kaldi/pull/1972/files#diff-2> (5)
> > > - *D* tools/extras/install_atlas.sh
> > > <https://github.com/kaldi-asr/kaldi/pull/1972/files#diff-3> (54)
> > > - *D* tools/install_atlas.sh
> > > <https://github.com/kaldi-asr/kaldi/pull/1972/files#diff-4> (1)
> > >
> > > Patch Links:
> > >
> > > - https://github.com/kaldi-asr/kaldi/pull/1972.patch
> > > - https://github.com/kaldi-asr/kaldi/pull/1972.diff
> > >
> > > —
> > > You are receiving this because you were mentioned.
> > > Reply to this email directly, view it on GitHub
> > > <https://github.com/kaldi-asr/kaldi/pull/1972>, or mute the thread
> > > <https://github.com/notifications/unsubscribe-auth/
> > ADJVu0HcP80bll5EDix0Bw554S0N-OWRks5svtz7gaJpZM4QFhjO>
> > > .
> > >
> >
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/kaldi-asr/kaldi/pull/1972#issuecomment-339410451>,
> or mute
> > the thread
> > <https://github.com/notifications/unsubscribe-
> auth/AEi_ULUFbQ-VfDWEtelovF-SHMzMDL9Mks5sv3IKgaJpZM4QFhjO>
> > .
> >
>
>
>
> --
> Daniel Galvez
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/1972#issuecomment-339424037>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ADJVuwnWWrdz990eMkwPyP_dX6hV6Y8_ks5sv3xWgaJpZM4QFhjO>
> .
>

galv commented:
See #1974
